addons/knobs/src/components/types/Array.js

```diff
@@ -10,28 +10,18 @@ function formatArray(value, separator) {
   return value.split(separator);
 }
 
-class ArrayType extends React.Component {
```

Comment:
I'm not sure how useful `PureComponent` is here. 

Since `knobs` is a object, shallow comparisons done by `PureComponent` are always going to return `false`. and maybe will even have a slight performance hit since we're always doing a shallow comparison of prop values

Reply:
I mean that using `PureComponent` is unnecessary since the shallow comparison of props (in this case `this.props.knob`) is always going to return `false`

Thus they should remain `Component`

There is no need for an implementation of `shouldComponentUpdate`